cmd/dep/testdata/harness_tests/ensure/add/all-new-spec/final/Gopkg.toml

```diff
@@ -5,4 +5,4 @@
 
 [[constraint]]
   name = "github.com/sdboyer/deptest"
-  version = "0.8.1"
+  version = "=0.8.1"
```

Comment:
The original requirements for #1036 also wanted to preserve recording a `^` in Gokpkg.toml.

> now, when a semver version is passed, we still want to ultimately record the caret in Gopkg.toml. but we make that change only after the solve works, so that we've got the desired version in Gopkg.lock.

Reply:
Might need @sdboyer to weigh in here. Recording the `^` would be the same as leaving it as an implicit caret.

Reply:
we definitely want the behavior of `^` here, not `=`, as @carolynvs notes. 

we're already doing some potentially confusing, frustrating magic on the user by having implicit carets. given the explicitness of the user input here, it's probably better to be explicit that the system is also coercing to caret. so, explicit caret here.

Reply:
aha! yes I misunderstood. I thought you meant that `@1.0.0` on cli -> `=1.0.0` in toml. 👍 will rework this.

Reply:
ah yeah, nope - the essence of this is the recognition that _explicitly_ specifying ranges on the CLI isn't super-useful. what the user generally wants is "i want this version right now." so we narrow what the CLI supports to only being explicit versions (so that the syntax there is simpler), and fulfill that "i want this version right now" goal. 

but, we don't necessarily have to interpret it as "[i want this version right now, and] only this version for ever and ever", which is what `=` means. `^` means "[i want this version right now, and] it's OK to upgrade later".